### ISSUE TITLE
Missing output

### ISSUE THREAD

**Issue by mgaudet** [rubyomr-preview/ruby#15]
```ruby
# test_from_prime_division.rb
require 'prime'

def assert_equal expected, actual, msg=""
   if expected == actual  
      puts caller_locations[0] 
   else 
      puts "error! #{caller_locations[0]}" 
      puts "#{expected} != #{actual}" 
      raise "error"
   end
end

class Foo 
# The first 100 prime numbers
  PRIMES = [
    2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37,
    41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83,
    89, 97, 101, 103, 107, 109, 113, 127, 131,
    137, 139, 149, 151, 157, 163, 167, 173, 179,
    181, 191, 193, 197, 199, 211, 223, 227, 229,
    233, 239, 241, 251, 257, 263, 269, 271, 277,
    281, 283, 293, 307, 311, 313, 317, 331, 337,
    347, 349, 353, 359, 367, 373, 379, 383, 389,
    397, 401, 409, 419, 421, 431, 433, 439, 443,
    449, 457, 461, 463, 467, 479, 487, 491, 499,
    503, 509, 521, 523, 541,
  ]

  def test_from_prime_division
    assert_equal PRIMES.inject(&:*),  Integer.from_prime_division(PRIMES.map{|p| [p,1]})
    assert_equal(-PRIMES.inject(&:*), Integer.from_prime_division([[-1, 1]] + PRIMES.map{|p| [p,1]}))
  end

end

Foo.new.test_from_prime_division
```
We see that any output is lost here!
```
$ OMR_JIT_OPTIONS=-Xjit:count=0,limit={*prime.rb:172:int_from_prime_division*} LD_LIBRARY_PATH=$PWD  ./ruby --disable-gems -Ilib test_from_prime_division.rb
$ ./ruby --disable-gems -Ilib test_from_prime_division.rb 
test_from_prime_division.rb:30:in `test_from_prime_division'
test_from_prime_division.rb:31:in `test_from_prime_division'
```

**Comment by mgaudet** [comment]
Investigating this I'm starting to suspect that the culprit might be poor interaction between a JITted method and `forwardable.rb` -- A reduced test case (below) that bypasses `primes` and instead calls `s int_from_prime_division` directly succeeds

```ruby
def int_from_prime_division(pd)
    pd.inject(1){|value, (prime, index)|
      value * prime**index
    }
end

primes = [[2,1]]
puts int_from_prime_division(primes)
```

Yet, stripping out the forwarding and call through `Integer`, we see the bug again: 

```ruby
require 'prime'

primes = [[2,1]]
puts Integer.from_prime_division(primes)
```

**Comment by mgaudet** [comment]
Progress on reducing the test case. This removes the reliance on `primes` 
```ruby
require "forwardable"
require "singleton"

class Prime
  include Singleton

  class << self
    extend Forwardable
    def method_added(method) # :nodoc:
      (class<< self;self;end).def_delegator :instance, method
    end
  end


  def int_from_prime_division(pd) # When this method is compiled, there is no output!
    pd.inject(1){|value, (prime, index)|
      value * prime**index
    }
  end
end


primes = [[2,1]]
puts Prime.int_from_prime_division(primes)
```

**Comment by mgaudet** [comment]
Binary searching `lib/forwardable.rb` points the finger at 0f11cda48ca73fdcdbbe9f72e185d017d218d196, which changed the implementation to avoid a `__send__`.

I don't claim to understand what's going at all just yet, but this is where to start looking.

**Comment by mgaudet** [comment]
(This [also caused issues for JRuby](https://bugs.ruby-lang.org/issues/12938). Need to investigate if the patch in that issue (r56848) also fixes the JIT)

**Comment by mgaudet** [comment]
Resolved by r56848

I will close this issue once I've submitted a PR updating `ruby_2_4_omr`.

**Comment by mgaudet** [comment]
Resolution isn't quite right: Update hides the issue, but it remains. Will update more on that shortly.

**Comment by mgaudet** [comment]
Closed by Pull Request #17
